mon-pix/app/routes/index.js

```diff
@@ -15,7 +15,11 @@ export default Route.extend(UnauthenticatedRouteMixin, {
 
           if (connectedUser.get('organizations.length')) {
             this.transitionTo('board');
-          } else {
+          }
+          if (connectedUser.get('usesProfileV2')) {
+            return this.replaceWith('profilv2');
```

Comment:
pourquoi `this.transitionTo('board')` et `this.transitionTo('compte')` n'ont pas de `return` ? Rajouter un `return` ici va changer quoi ?
Peut-être y a-t-il une bonne raison au fait de ne pas `return` ?

Reply:
Normalement, pas besoin de return avec les transitionTo, ça capte tout de suite l'evenement de la transition et ça sort de la fonction.
C'est une préférence de préciser le return pour assurer le retour

Reply:
et du coup, c'est `replaceWith` qui a été choisi ici, est-ce que c'est plus pertinent de mettre `replaceWith` que `transitionTo` ? Si oui, pourquoi ne pas changer les autres en `replaceWith` ?

Reply:
A priori `transitionTo`/`replaceWith` ne lance pas d'exception donc le code qui suit va être exécuté. Par exemple ici si `connectedUser.get('organizations.length')` et `connectedUser.get('usesProfileV2')` sont vrais tous les deux il y aura deux appels à `transitionTo` :scream:

Par ailleurs la valeur de retour de `transitionTo` est _thenable_ (même si ce n'est pas documenté tel quel…).

Tout ça me fait suggérer de mettre effectivement des `return` devant tous les `transitionTo` / `replaceWith`.

Reply:
Sinon, surcharger `beforeModel` de `UnauthenticatedRouteMixin` sans appeler `super()` c'est interdit : https://ember-simple-auth.com/api/classes/UnauthenticatedRouteMixin.html , est-ce qu'on pourrait pas en profiter pour corriger ça ? A minima en enlevant l'héritage de `UnauthenticatedRouteMixin` qui n'a finalement aucun effet, ou alors en surchargeant la propriété `routeIfAlreadyAuthenticated` au lieu de `beforeModel` ?

Reply:
J'ai nettoyer l'index, en supprimant le `UnauthenticatedRouteMixin`, et en supprimant aussi des appels qui étaient obsolètes liés à l'ancien affichage de l'index